retriever/lib/engine.py

```diff
@@ -406,19 +406,28 @@ def database_name(self, name=None):
 
     def download_file(self, url, filename):
         """Download file to the raw data directory."""
+        downloaded = False
         if not self.find_file(filename) or not self.use_cache:
             path = self.format_filename(filename)
             self.create_raw_data_dir()
             print("\nDownloading " + filename + "...")
             try:
                 urlretrieve(url, path, reporthook=reporthook)
+                downloaded = True
             except:
                 # For some urls lacking filenames urlretrieve from the future
                 # package seems to fail. This issue occurred in the PlantTaxonomy
                 # script. If this happens, fall back to the standard Python 2 version.
                 from urllib import urlretrieve as py2urlretrieve
-                py2urlretrieve(url, path, reporthook=reporthook)
+                try:
+                    py2urlretrieve(url, path, reporthook=reporthook)
+                except:
+                    pass
```

Comment:
Would replacing `print("Unable to download dataset {}".format(filename))`  for pass in line 425 , not be enough to catch the errors?

Reply:
if you make changes would you also update to current master.
1. update master
`git fetch upstream`
`git merge upstream master`


2. In your current branch
`git rebase master`

Reply:
If I replace line 425 with ```print("Unable to download dataset {}".format(filename))``` it just prints ```cannot import name 'urlretrieve'``` and not the message.